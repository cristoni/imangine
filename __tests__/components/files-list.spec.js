import React from 'react';
import FilesList from '../../components/files-list';

test('FilesList renders the uploaded files list', () => {
  let wrapper = shallow(<FilesList />);
  expect(wrapper).toMatchSnapshot();

  const mockFiles = [
    { name: 'file1' },
    { name: 'file2' },
    { name: 'file3' }
  ];

  wrapper = shallow(<FilesList files={mockFiles} />);
  expect(wrapper).toMatchSnapshot();

  const listGroupItems = wrapper.find('.list-group-item');
  expect(listGroupItems.length).toBe(mockFiles.length);
});