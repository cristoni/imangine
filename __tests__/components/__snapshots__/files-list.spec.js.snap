// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesList renders the uploaded files list 1`] = `
<div
  className="list-group shadow"
/>
`;

exports[`FilesList renders the uploaded files list 2`] = `
<div
  className="list-group shadow"
>
  <a
    className="list-group-item list-group-item-action"
    href="#"
    key="0"
  >
    file1
  </a>
  <a
    className="list-group-item list-group-item-action"
    href="#"
    key="1"
  >
    file2
  </a>
  <a
    className="list-group-item list-group-item-action"
    href="#"
    key="2"
  >
    file3
  </a>
</div>
`;
